ofrak_core/ofrak/service/data_service.py

```diff
@@ -134,6 +146,33 @@ async def delete_models(self, data_ids: Iterable[DataId]) -> None:
             root.delete_mapped_model(model)
             del self._model_store[model.id]
 
+    async def search(self, data_id, query, start=None, end=None, max_matches=None):
+        model = self._get_by_id(data_id)
+        root = self._get_root_by_id(model.root_id)
+        start = model.range.start if start is None else model.range.start + start
+        end = model.range.end if end is None else min(model.range.end, model.range.start + end)
+        if type(query) is bytes:
```

Comment:
Support for subclassing is more robust to future changes.

```suggestion
        if isinstance(query, bytes):
```